### PR TITLE
[CELEBORN-1477][FOLLOWUP] Fix api v1 response issue

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

1. Fix below api response:

- master GET /api/v1/masters
- master GET /api/v1/applications/top_disk_usages
- master&worker /api/v1/thread_dump

2. Fix typo in migration guide

3. refine the api annotation: METHOD -> PATH

4. enhance the `RestExceptionMapper`
### Why are the changes needed?

For /api/v1/masters, the `id` field is not in good format.
```
{
"groupId": "c5196f6d-2c34-3ed3-8b8a-47bede733167",
"leader": {
"id": "<ByteString@4960c29e size=1 contents=\"0\">",
"address": "...:9872"
},
...
}
```

For `/api/v1/applications/top_disk_usages`, it thrown NPE, we shall filter the null items.
```
24/07/18 21:52:38,506 WARN [master-JettyThreadPool-40] RestExceptionMapper: Error occurs on accessing REST API.
java.lang.NullPointerException
	at org.apache.celeborn.service.deploy.master.http.api.v1.ApplicationResource.$anonfun$topDiskUsedApplications$2(ApplicationResource.scala:78)
```

For `api/v1/thread_dump`, seems need to add `@Produces(Array(MediaType.APPLICATION_JSON))`:
```
Caused by: javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
	at org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundWriteTo(MappableExceptionWrapperInterceptor.java:65)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1116)
	at org.glassfish.jersey.server.ServerRuntime$Responder.writeResponse(ServerRuntime.java:649)
	at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:380)
	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:426)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:264)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:235)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
	... 36 more
Caused by: org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException: MessageBodyWriter not found for media type=text/html, type=class scala.collection.immutable.Map$Map1, genericType=class scala.collection.immutable.Map$Map1.
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:224)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)
	at org.glassfish.jersey.server.internal.JsonWithPaddingInterceptor.aroundWriteTo(JsonWithPaddingInterceptor.java:85)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:139)
	at org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundWriteTo(MappableExceptionWrapperInterceptor.java:61)
	... 51 more
```

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

Integration testing.

For `api/v1/masters`:
image

For master `api/v1/applications/top_disk_usages`:
image

For `api/v1/thread_dump`:
image

### CODE REVIEW

**Comment by turboFei** [comment]
cc @pan3793

**Comment by turboFei** [comment]
cc @pan3793

**Review by cxzl25 — APPROVED** [review]

File: service/src/main/scala/org/apache/celeborn/server/common/http/api/v1/ApiV1BaseResource.scala [hunk 1]
```diff
@@ -51,7 +52,7 @@ class ApiV1BaseResource extends ApiRequestContext {
           .threadName(threadStack.threadName)
           .threadState(threadStack.threadState.toString)
           .stackTrace(threadStack.stackTrace.elems.asJava)
-          .blockedByThreadId(threadStack.blockedByThreadId.getOrElse(null.asInstanceOf[Long]): Long)
+          .blockedByThreadId(threadStack.blockedByThreadId.getOrElse(null).asInstanceOf[java.lang.Long])
```
Comment (turboFei) [inline service/src/main/scala/org/apache/celeborn/server/common/http/api/v1/ApiV1BaseResource.scala:55]:
```
scala> null.asInstanceOf[Long]
val res2: Long = 0

scala> None.getOrElse(null).asInstanceOf[java.lang.Long]
val res3: Long = null

```

**Comment by turboFei** [comment]
thanks, merging to master

**Comment by turboFei** [comment]
thanks, merged to main

**Pull request closed without merging**